pkg/infra/teams/team3/helpers.go

```diff
@@ -156,3 +204,31 @@ func minMax4(isMax bool, nums [4]float64) float64 {
 	}
 	return ans
 }
+
+func (a *AgentThree) getGroupAvStats(baseAgent agent.BaseAgent) (float64, float64, float64, float64) {
+	groupAvHP := AverageArray(GetHealthAllAgents(baseAgent))
+	groupAvST := AverageArray(GetStaminaAllAgents(baseAgent))
+	groupAvATT := AverageArray(GetAttackAllAgents(baseAgent))
+	groupAvDEF := AverageArray(GetDefenceAllAgents(baseAgent))
+
+	return groupAvHP, groupAvST, groupAvATT, groupAvDEF
```

Comment:
similar to before - worth wrapping in `groupAv struct`

Reply:
Or, indeed, abstracting to a `avg struct` for all of these functions

Reply:
created an avgerageStats struct and a ThresholdVal struct for the two multi-var returns